source/MaterialXView/Viewer.cpp

```diff
@@ -1415,7 +1415,11 @@ void Viewer::saveShaderSource(mx::GenContext& context)
         mx::TypedElementPtr elem = material ? material->getElement() : nullptr;
         if (elem)
         {
-            mx::ShaderPtr shader = createShader(elem->getNamePath(), context, elem);
+            const bool hasTransparency = mx::isTransparentSurface(elem, context.getShaderGenerator().getTarget());
+            mx::GenContext materialContext = context;
+            materialContext.getOptions().hwTransparency = hasTransparency;
+
+            mx::ShaderPtr shader = createShader(elem->getNamePath(), materialContext, elem);
```

Comment:
This code path was missing the check for transparency, so saved GLSL shaders would never include transparency handling.